util/test_download_script.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# Define the path to your Python script
+PYTHON_SCRIPT="download_util_script.py"
+
+# --- Test Case 1: Successful Download (mandatory flags provided) ---
+echo "--- Running Test Case 1: Successful Download ---"
+TEST_URL="https://www.stats.govt.nz/assets/Topics/Population/Summary-figures-for-the-NZ-population-1991-2023.xlsx"
+TEST_OUTPUT_FOLDER="./test_downloads/case1"
+
+
+python3 "${PYTHON_SCRIPT}" \
+  --url="${TEST_URL}" \
+  --output_folder="${TEST_OUTPUT_FOLDER}" \
+
+if [ $? -eq 0 ]; then
```

Comment:
test should check if the output file was created fresh, i.e., delete existing output before calling download and check if output file has sufficient content  or is not empty  after download.